packages/medusa/src/subscribers/search.ts

```diff
@@ -0,0 +1,94 @@
+import EventBusService from "../services/event-bus"
+import { SEARCH_INDEX_EVENT } from "../loaders/search-index"
+import ProductService from "../services/product"
+import { indexTypes } from "medusa-core-utils"
+import { SearchService } from "../services"
+import { Product } from "../models"
+
+type InjectedDependencies = {
+  eventBusService: EventBusService
+  searchService: SearchService
+  productService: ProductService
+}
+
+class BatchJobSubscriber {
+  private readonly eventBusService_: EventBusService
+  private readonly searchService_: SearchService
+  private readonly productService_: ProductService
+
+  constructor({
+    eventBusService,
+    searchService,
+    productService,
+  }: InjectedDependencies) {
+    this.eventBusService_ = eventBusService
+    this.searchService_ = searchService
+    this.productService_ = productService
+
+    this.eventBusService_.subscribe(SEARCH_INDEX_EVENT, this.indexDocuments)
+  }
+
+  indexDocuments = async (): Promise<void> => {
+    const TAKE = 20
```

Comment:
thought: We could probably increase this quite a bit - maybe 500?

Reply:
>When using the [saveObjects](https://www.algolia.com/doc/api-reference/api-methods/save-objects/) method, the API client automatically chunks your records into batches of 1,000 objects.

We can probably goes up to 1000 wdyt?